sql/catalyst/src/main/scala/org/apache/spark/sql/types/Decimal.scala

```diff
@@ -134,7 +135,14 @@ final class Decimal extends Ordered[Decimal] with Serializable {
   def set(bigintval: BigInteger): Decimal = {
     try {
       this.decimalVal = null
-      this.longVal = bigintval.longValueExact()
+      // TODO: Remove this once we migrate to java8 and use longValueExact() instead.
+      val isExactLong = bigintval.compareTo(BigInteger.valueOf(JLong.MAX_VALUE)) <= 0 &&
+        bigintval.compareTo(BigInteger.valueOf(JLong.MIN_VALUE)) >= 0
+      if(isExactLong) {
```

Comment:
Space after if. Don't throw an exception just to catch it; throw the exception below here. No need for "else"